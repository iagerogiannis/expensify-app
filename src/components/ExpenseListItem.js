import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link className="list-item" to={`/edit/${id}`}>
        <div className="list-item__expense">
            <h3>{description}</h3>
            <p>{moment(createdAt).format('MMMM Do, YYYY')}</p>
        </div>
        <h3 className="list-item__amount">{numeral(amount / 100).format('$0,0.00')}</h3>
    </Link>
);

export default ExpenseListItem;
