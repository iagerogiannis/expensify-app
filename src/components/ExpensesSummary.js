import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import 'numeral/locales/it';

numeral.locale('it');

export const ExpensesSummary = ({ expenses }) => {

    const expenseCount = expenses.length;
    const expensesTotal = getExpensesTotal(expenses);
    const expenseWord = expenses.length === 1 ? 'expense' : 'expenses';
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('0,0.00 $');

    return (
        <p>Viewing {expenseCount} {expenseWord} totalling {formatedExpensesTotal}</p>
    )
};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
