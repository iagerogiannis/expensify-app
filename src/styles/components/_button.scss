.button {
  background-color: $sky-blue;
  color: white;
  border: none;
  transition: transform 0.2s;
  font-size: $font-large;
  font-weight: 300;
  padding: $s-size;
  text-decoration: none;
  display: inline-block;
  line-height: 1;
  min-width: 18rem;
  text-align: center;
}

.button:hover {
  transform: scale(1.075);
}

.button:active {
  transition: transform 0s;
  transform: scale(0.95);
}

.button--secondary {
  background-color: #888;
}
