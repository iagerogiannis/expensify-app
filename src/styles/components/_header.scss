.header {
  background-color: $dark-blue;
}

.header__content {
  align-self: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

.header__title {
  color: white;
  text-decoration: none;
  h1 {
    margin: 0;
  }
}

.header__button {
  background-color: transparent;
  border: none;
  color: white;
  font-weight: 100;
}
