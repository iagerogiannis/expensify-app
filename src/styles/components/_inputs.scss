.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  font-size: $font-large;
  font-weight: 300;
  padding: $s-size;
  resize: vertical;
}

.select {
  @extend .text-input;
}

.text-area {
  @extend .text-input;
  height: 10rem;
}
