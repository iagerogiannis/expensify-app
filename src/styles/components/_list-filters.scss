.list-filters {
  display: flex;
  flex-direction: column;
  margin-bottom: $m-size;

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    margin-bottom: $l-size;
  }
}

.list-filters__item {
  margin-bottom: $s-size;
  width: 336px;
  > * {
    width: 100%;
  }

  @media (min-width: $desktop-breakpoint) {
    width: auto;
    margin: 0 $s-size 0 0;
  }
}
