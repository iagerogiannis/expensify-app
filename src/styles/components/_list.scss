.list-header {
  background-color: $off-white;
  border: 1px solid darken($off-white, 7%);
  color: $grey;
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
}

.list-item {
  border: 1px solid darken($off-white, 7%);
  border-top: none;
  display: flex;
  flex-direction: column;
  padding: $s-size $m-size;
  text-decoration: none;
  > * {
    color: black;
    margin: 0;
  }
  transition: background-color 0.3s ease;
  &:hover {
    background-color: lighten($off-white, 2%);
  }
  @media (min-width: $desktop-breakpoint-2) {
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }
}

.list-item__expense {
  word-break: break-all;
  flex-direction: column;
  > * {
    margin: 0;
  }
  h3 {
    margin-top: $xs-size;
  }
  p {
    color: $grey;
    font-size: $font-small;
  }
  @media (min-width: $desktop-breakpoint-2) {
    padding-right: 10rem;
    p {
      margin-bottom: $xs-size;
    }
  }
}

.list-item__amount {
  color: black;
  padding: $s-size 0 0 0;
  @media (min-width: $desktop-breakpoint-2) {
    padding: $s-size 0;
  }
}

.list-item--message {
  margin: 0;
  border: 1px solid darken($off-white, 7%);
  border-top: none;
  padding: $s-size $m-size;
  font-style: italic;
  color: $grey;
  text-align: center;
}
