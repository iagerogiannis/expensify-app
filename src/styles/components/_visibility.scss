.show-for-mobile {
  @media (min-width: $desktop-breakpoint-2) {
    display: none;
  }
}

.show-for-desktop {
  @media (max-width: $desktop-breakpoint-2) {
    display: none;
  }
}
