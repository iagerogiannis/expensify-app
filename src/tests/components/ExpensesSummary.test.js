import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should correctly render ExpensesSummary for 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});
