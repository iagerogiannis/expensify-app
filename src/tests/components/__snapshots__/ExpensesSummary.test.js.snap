// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary for 1 expense 1`] = `
<h1>
  Viewing 
   
  expenses
   totalling 
  0,00 €
</h1>
`;

exports[`should correctly render ExpensesSummary for multiple expenses 1`] = `
<h1>
  Viewing 
   
  expenses
   totalling 
  0,00 €
</h1>
`;
